public/components/trace_analytics/components/common/shared_components/custom_datagrid.tsx

```diff
@@ -88,36 +108,126 @@ export const RenderCustomDataGrid: React.FC<RenderCustomDataGridParams> = ({
+  const tableModeSelector =
+    setTracesTableMode && tracesTableMode ? (
+      <EuiPopover
+        isOpen={isPopoverOpen}
+        closePopover={() => setIsPopoverOpen(false)}
+        panelStyle={{ width: '350px' }}
+        button={
+          <EuiFlexGroup alignItems="center" gutterSize="s" direction="row">
+            <EuiFlexGroup alignItems="center" gutterSize="xs" direction="row">
+              <EuiFlexItem grow={false}>
+                <EuiText size="s">
+                  <EuiTextColor
+                    data-test-subj="trace-table-mode-selector"
+                    color="success"
+                    onClick={() => setIsPopoverOpen(!isPopoverOpen)}
+                    style={{ cursor: 'pointer', display: 'flex', alignItems: 'center' }}
+                  >
+                    {TRACE_TABLE_TITLES[tracesTableMode]} ({displayedRowCount})
+                    <EuiIcon type="arrowDown" size="s" color="success" style={{ marginLeft: 4 }} />
+                  </EuiTextColor>
+                </EuiText>
+              </EuiFlexItem>
+            </EuiFlexGroup>
+            <EuiFlexItem grow={false} />
+          </EuiFlexGroup>
+        }
       >
-        {isFullScreen
-          ? i18n.translate('toolbarControls.exitFullScreen', {
-              defaultMessage: 'Exit full screen',
-            })
-          : i18n.translate('toolbarControls.fullScreen', {
-              defaultMessage: 'Full screen',
-            })}
-      </EuiButtonEmpty>,
-      ...toolbarButtons,
-    ],
-    [isFullScreen, toolbarButtons]
-  );
+        <EuiPopoverTitle>Select trace table filter</EuiPopoverTitle>
+        <EuiSelectable
+          singleSelection="always"
+          options={tableOptions}
+          listProps={{ rowHeight: 80 }}
+          renderOption={(option, searchValue) => (
+            <div style={{ padding: '6px 10px', width: '100%' }}>
+              <EuiHighlight search={searchValue}>{TRACE_TABLE_TITLES[option.key]}</EuiHighlight>
+              <br />
+              <EuiTextColor color="subdued" className="popOverSelectableItem">
+                <small>{option['aria-describedby'] || 'No description available'}</small>
+              </EuiTextColor>
+            </div>
+          )}
+          onChange={(newOptions) => {
+            const selectedMode = newOptions.find((option) => option.checked === 'on')?.key;
+            if (selectedMode && selectedMode !== tracesTableMode) {
+              setTracesTableMode(selectedMode);
+              sessionStorage.setItem(TRACE_TABLE_TYPE_KEY, selectedMode);
+            }
+            setIsPopoverOpen(false);
+          }}
+        >
+          {(list) => list}
+        </EuiSelectable>
+      </EuiPopover>
+    ) : null;
+
+  const toolbarControls = useMemo(() => {
+    const controls = [];
+
+    if (tableModeSelector) {
+      controls.push(tableModeSelector);
+    }
+
+    if (!['all_spans', 'root_spans', 'service_entry_spans'].includes(tracesTableMode ?? '')) {
```

Comment:
can simplify this to `if (tracesTableMode === traces)`

Reply:
Switched too 
```
if (tracesTableMode === 'traces' || tracesTableMode === undefined) {
```
Using undefined for the tables made inside tracesView and flyout